### ISSUE TITLE
Use 'set -o errexit' to catch script errors

### ISSUE THREAD

**Issue by hreinecke** [osandov/blktests#89]
Heya,

bash has the option 'errexit' (invoked via 'set -o errexit'), which will cause the script to abort if a command/line returns 'false'.
This has the nice effect that any errors are catched automatically, and don't have to be checked for.
With that we'll be able to catch silent failures which are currently ignored.
Drawback is that one gets lots of false positive initially, as every command which _should_ be ignored need to be suffixed by ' || true'.

But it's a very good thing for hardening the scripts; I've had quite good experience with that when developing testcases for [md_monitor](https://github.com/hreinecke/md_monitor)

**Comment by kawasaki** [comment]
As I noted at LSF 2023, I think this is a good idea. Yesterday, I found time to work on this, and did a trial implementation: a99d840248a1. With this change, I observed many test cases fail or show weird behavior as @hreinecke indicated. It looks like certain amount of work is required to fix them all. I will spend some more time and see how much work will be required.

**Comment by bvanassche** [comment]
Is `set -o errexit` equivalent to `set -e`? Do others have a preference for one of these two alternatives? I slightly prefer `set -e`. `set -e` already occurs in the file `new`.

**Comment by kawasaki** [comment]
> Is `set -o errexit` equivalent to `set -e`? Do others have a preference for one of these two alternatives? I slightly prefer `set -e`. `set -e` already occurs in the file `new`.

Thanks for the comment. I will user `set -e`.

**Comment by kawasaki** [comment]
I've spent some time on the errexit support. The number of affected test cases are not so small, but it looks possible to modify them to meet the errexit requirements. Around a dozen of patches will be required. WIP code is available [at my errexit branch](https://github.com/kawasaki/blktests/commits/errexit).

Through this work, I noticed following points:

1. Some commands in test cases need to return non-zero error status to meet the test purpose. For those commands, we need to add "|| true". This looks a bit weird and I'm not sure if blktests contributors will like it or not. Some discussion on the list will be needed.
2. If status check of commands connected with pipes, some complicated changes will be required. scsi/006 is the test case with such a pipe. This adds some complexity, but I think such cases are rare and should be fine.
3. With current implementation, some of clean up operations after each test can be skipped when errexit happens in the test case. This is not good. It is desired to improve clean up in the blktests framework before errexit introduction.
4. I found that some of the commands in nbd/002 and nbd/003 return non-zero status, and not sure if these are expected or not. Will need further work. 

In short, this errexit support work needs larger effort than I expected. I will rethink priority of this work. I think nbd/002,003 improvement and clean-up improvement will have higher priority.

**Comment by bvanassche** [comment]
How about using this alternative for ignoring an error status:
```
if command_that_may_fail; then :; fi
```

**Comment by kawasaki** [comment]
> How about using this alternative for ignoring an error status:
> 
> ```
> if command_that_may_fail; then :; fi
> ```

That will work too, and could be easier to understand for some people :) One drawback is that this needs more types than "|| true".

This made me think of creating a new helper function like _ok() or _ignore_status(). With these "|| _ok" or "|| _ignore_status", and these might be a bit easier to understand than "|| true".

**Comment by kawasaki** [comment]
This topic was shortly discussed in LSF 2024 blktests session. I reported that the script changes are required to support errexit. Omar noted that he did not enabled errexit by design. IIRC, he did so to avoid the weird script changes required. After that, I did not set priority on this work.

**Comment by kawasaki** [comment]
Recently, @bvanassche [posted a patch for a new test case](https://lore.kernel.org/linux-block/ckctv7ioomqpxe2iwcg6eh6fvtzamoihnmwxvavd7lanr4y2y6@fbznem3nvw3w/), and used "set -e" in the test case. And the discussion has been started. The new test case does not have weird way for the errexit support. So I came up an idea to enable "set -e" for specific set of test cases. Based on this idea, I [posted a patch series](https://lore.kernel.org/linux-block/20250530075425.2045768-1-shinichiro.kawasaki@wdc.com/) as RFC.
